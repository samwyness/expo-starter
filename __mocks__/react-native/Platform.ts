export const mockPlatform = (os: string, version?: string | number) => {
  jest.doMock('react-native/Libraries/Utilities/Platform', () => ({
    ...jest.requireActual('react-native/Libraries/Utilities/Platform'),
    OS: os,
    select: (osList: Record<string, unknown>) => osList[os],
    Version: version,
  }));
};
