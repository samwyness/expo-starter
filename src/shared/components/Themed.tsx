/**
 * Learn more about Light and Dark modes:
 * https://docs.expo.io/guides/color-schemes/
 */

import { ComponentProps } from 'react';
import { Text as DefaultText, View as DefaultView } from 'react-native';

import { useThemeColor } from '../hooks/useThemeColor';

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

export type TextProps = ThemeProps & ComponentProps<typeof DefaultText>;
export type ViewProps = ThemeProps & ComponentProps<typeof DefaultView>;

export function Text(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return <DefaultText style={[{ color }, style]} {...otherProps} />;
}

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    'background',
  );

  return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
}
