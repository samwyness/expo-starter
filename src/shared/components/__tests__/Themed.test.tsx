import { expect, it } from '@jest/globals';
import { render } from '@testing-library/react-native';

import { Text, View } from '../Themed';

it(`renders Text correctly`, () => {
  // Arrange
  const tree = render(<Text>Some Text</Text>).toJSON();

  // Act

  // Assert
  expect(tree).toMatchSnapshot();
});

it(`renders View correctly`, () => {
  // Arrange
  const tree = render(
    <View lightColor="#FFFFFF" darkColor="#000000" />,
  ).toJSON();

  // Act

  // Assert
  expect(tree).toMatchSnapshot();
});
