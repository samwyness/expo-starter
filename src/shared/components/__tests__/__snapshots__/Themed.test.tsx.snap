// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Text correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Some Text
</Text>
`;

exports[`renders View correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      undefined,
    ]
  }
/>
`;
